### DISCUSSION TITLE
[InternalZoneSSLCert] Failure deploing Hubs cloud via AWS

### DISCUSSION THREAD

**Discussion by Powerstephan** [mozilla/hubs#5549] [Hubs Cloud Questions]
**Description**
If i try to deploy Hubs Cloud via AWS (Using Route 53 Domain) everything performs good, untill SSL Certificate Creation.
The the process cancels prompting the following error:
<html>
<body>

2022-06-30 10:16:24 UTC+0200 | ImmBlend-Hubs-Metaverse | CREATE_FAILED | The following resource(s) failed to create: [InternalZoneSSLCert].
-- | -- | -- | --
2022-06-30 10:16:21 UTC+0200 | InternalZoneSSLCert | CREATE_FAILED | Received response status [FAILED] from custom resource. Message returned: Certificate not issued in time (RequestId: b1b9cd53-1e08-484d-9232-95b80a137a5c)

</body>
</html>

Bevor Updating i deleted all CNAME entries, but this does not solve the problem :(

**Expected behavior**
I expected to deploy the hubs cloud correct ;)

**Screenshots**
attached

![unknown](https://user-images.githubusercontent.com/105124887/176678012-11ca4488-ae2f-4e18-94f4-dce055a4069d.png)

**Comment by yukipioa** [comment]
@Powerstephan 
I am having the same problem.
I don't see any reply yet, has it been resolved?

**Reply by pattersonbl2** [reply]
Are the domains hosted in Route53?   Also, I would look at errors related to AWS  cert management.  The C name shouldn't be the cause of this.

**Reply by Powerstephan** [reply]
I Don't Know how, but no it Works fine. But just the Personal Edition. :/

**Reply by Powerstephan** [reply]
> Are the domains hosted in Route53?   Also, I would look at errors related to AWS  cert management.  The C name shouldn't be the cause of this. 

Yes bought and Hostel via Route53

**Reply by pattersonbl2** [reply]
I created a new stack using the multi-server setup without any problems.  How are you entering the domain name in the stack?  This could be a wide range of issues related to the account, not the hub's product.  Your domain shouldn't have any names if the stack didn't finish, Did you use this domain before on another project?

**Reply by yukipioa** [reply]
In my case, I have previously deployed Hubs Cloud personal using that domain as a test environment.

I wanted to change to Enterprise using that domain, so I deleted the stack and re-created it.
Is it not possible to use the domain that was used before again?

**Reply by Powerstephan** [reply]
the typing of the domain was correct. The same procedure works well with personal cloud, but not with the enterprise.

Now, the personal is runnung, but the avatar system is not working. Realy bad :(

**Reply by pattersonbl2** [reply]
Since we can't reproduce the issue I am at a loss for what it could be.   

I would suggest deleting your stack completely and redeploying a new hubs cloud stack using version 1.1.4. 
I would also not use the same stack name as before because that could cause issues if any resources are left over from previous stacks.

**Reply by pattersonbl2** [reply]
> Yes you are I have done it many times now.  
I would be mindful not to give your new stack the same name as the previous one because that could cause issues.

**Reply by TheNFTDAO** [reply]
Hi I have the same issue with both multi server and personal hubs cloud, everything was working fine up until 5 days ago when it suddenly gave errors when visiting any admin pages or rooms we host. I've been working with hubs for 7 months and we have had the same set up the whole time:

Domain - metac.(domain1).net
Internal - (domain1).net
Short link - (domain1).link

There are no leftover resources as I go through and delete any leftover certificates, ec2 instances, route 53 records and ssh keys and recreate them with different names but still it doesn't deploy.

I've tried different subdomains. I've deleted the cloudformation stack and manually deleted all route 53 records and tried deleting and recreating hosted zones from which to use hubs cloud but it always rolls back at the exact point as the original poster. The attached happens no matter what I use for domains, all domains were bought and are hosted through route 53.
![image](https://user-images.githubusercontent.com/99777666/179078498-b9f72b75-9fa8-4dd4-8fe2-c7e5056dc8b3.png)

**Reply by TheNFTDAO** [reply]
> Hi I have the same issue with both multi server and personal hubs cloud, everything was working fine up until 5 days ago when it suddenly gave errors when visiting any admin pages or rooms we host. I've been working with hubs for 7 months and we have had the same set up the whole time:
> 
> Domain - metac.(domain1).net Internal - (domain1).net Short link - (domain1).link
> 
> There are no leftover resources as I go through and delete any leftover certificates, ec2 instances, route 53 records and ssh keys and recreate them with different names but still it doesn't deploy.
> 
> I've tried different subdomains. I've deleted the cloudformation stack and manually deleted all route 53 records and tried deleting and recreating hosted zones from which to use hubs cloud but it always rolls back at the exact point as the original poster. The attached happens no matter what I use for domains, all domains were bought and are hosted through route 53. ![image](https://user-images.githubusercontent.com/99777666/179078498-b9f72b75-9fa8-4dd4-8fe2-c7e5056dc8b3.png)

I just changed my setup to the following:
domain - domain1.net
internal - domain2.link
shortlink - domain3.link

all domains hosted on route 53, all 3 separate domains with no records besides the NS and SOA records when I deploy. new cloudformation stack name

and now I get the following:
![image](https://user-images.githubusercontent.com/99777666/179148979-30da7469-f5e5-4548-bcb6-d6bf546844db.png)

The cloudformation rolls back and deletes everything. I go in and manually delete all certificates, terminate ec2 instances, and delete all records in route 53 and still I get this certificate not issued in time error. Please help

**Reply by TheNFTDAO** [reply]
went through the entire stack here and deleted everything related, trying again: https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/94ec4ef1-285d-4e63-b2eb-0b65ea93b942.01052d43-6397-498b-8a8f-7d64e91320b5.png

**Reply by TheNFTDAO** [reply]
> went through the entire stack here and deleted everything related, trying again: https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/94ec4ef1-285d-4e63-b2eb-0b65ea93b942.01052d43-6397-498b-8a8f-7d64e91320b5.png

this did not work, now it's the shortlinkzonessl error again.

**Reply by FrankAllanRJ** [reply]
The cause of the error ("InternalZoneSSLCert" - "Certificate not issued in time") is the delay (or non-occurrence) of the certificate owner's identity validation.

To confirm, go to "AWS Certificate Manager > Certificates" and see that in the list of certificates, one will have "Pending validation".

I still haven't figured out the reason for not validating, since the certificate was purchased and is hosted on Route 53. But I solved acquiring a new domain outside of AWS (by Namecheap).

**Comment by VesuviusGroup** [comment]
We had the same error with SSL certificates. Before deploying you have to register a SSL certificate for your sub domain (ex: hubs.yourdomain.com) in the AWS certificate manager. 
In the certificate listing will be a **Cname** name and **value**. You will have to **point your subdomain to those values**. This is to prove you have control over the domain. Wait until it gets validated, shouldn't take long.
After deploy you will have to change the Cname values to the ones provided by the Stack.
